src/classes/HiFiAudioAPIData.ts

```diff
@@ -66,33 +101,249 @@ export class OrientationEuler3D {
      * Construct a new `OrientationEuler3D` object. All parameters are optional. Unset parameters will be set to `0`. Remember, all units for member variables are `degrees`.
      */
     constructor({ pitchDegrees = 0, yawDegrees = 0, rollDegrees = 0 }: { pitchDegrees?: number, yawDegrees?: number, rollDegrees?: number } = {}) {
-        this.pitchDegrees = pitchDegrees;
-        this.yawDegrees = yawDegrees;
-        this.rollDegrees = rollDegrees;
+        this.pitchDegrees = clampNonan(pitchDegrees, -360, 360, 0);
```

Comment:
why do we want to _clamp_ these values? why wouldn't we want to add/subtract 360 degrees to ensure the values are within range?

Reply:
sounds good